src/prop/prop_proof_manager.cpp

```diff
@@ -300,12 +301,13 @@ bool PropPfManager::reproveUnsatCore(const std::unordered_set<Node>& cset,
       // dump using the CNF stream we created above
       csms.dumpDimacs(*outDimacs, aclauses);
     }
-    return true;
+    success = true;
   }
+  delete csm;
```

Comment:
rather than doing this, if you were to use a unique ptr when creating csm it'd solve the issue in a more principled manner maybe?